src/lib/statistics/chunk_statistics/counting_quotient_filter.cpp

```diff
@@ -0,0 +1,139 @@
+}
+
+template <typename ElementType>
+CountingQuotientFilter<ElementType>::~CountingQuotientFilter() {
+  boost::apply_visitor([&](auto& filter) {qf_destroy(&filter);}, _quotient_filter);
+}
+
+template <typename ElementType>
+void CountingQuotientFilter<ElementType>::insert(ElementType value, uint64_t count) {
+  uint64_t bitmask = static_cast<uint64_t>(std::pow(2, _hash_bits)) - 1;
+  uint64_t hash = bitmask & _hash(value);
+  for (size_t i = 0; i < count; i++) {
+    boost::apply_visitor([&](auto& filter) {qf_insert(&filter, hash, 0, 1);}, _quotient_filter);
+  }
+}
+
+template <typename ElementType>
+void CountingQuotientFilter<ElementType>::insert(ElementType value) {
+  insert(value, 1);
+}
+
+template <typename ElementType>
+uint64_t CountingQuotientFilter<ElementType>::count_all_type(AllTypeVariant value) const {
+  DebugAssert(value.type() == typeid(ElementType), "Value does not have the same type as the filter elements");
+  return count(type_cast<ElementType>(value));
+}
+
+template <typename ElementType>
+bool CountingQuotientFilter<ElementType>::can_prune(const AllTypeVariant& value,
+    const PredicateCondition predicate_type) const {
+  DebugAssert(predicate_type == PredicateCondition::Equals, "CQF only supports equality predicates");
+  return count_all_type(value) == 0;
+}
+
+template <typename ElementType>
+uint64_t CountingQuotientFilter<ElementType>::count(ElementType value) const {
+  uint64_t bitmask = static_cast<uint64_t>(std::pow(2, _hash_bits)) - 1;
+  uint64_t hash = bitmask & _hash(value);
+  uint64_t count;
+  boost::apply_visitor([&](auto& filter) {count = qf_count_key_value(&filter, hash, 0);}, _quotient_filter);
+  return count;
+}
+
+/**
+* Computes the hash for a value.
+**/
+template <typename ElementType>
+uint64_t CountingQuotientFilter<ElementType>::_hash(ElementType value) const {
+  auto hash = murmur2<ElementType>(value, _seed);
+  return static_cast<uint64_t>(hash);
+}
+
+/**
+* Computes the hash for a string.
+**/
+template <>
+uint64_t CountingQuotientFilter<std::string>::_hash(std::string value) const {
+  auto hash = murmur_hash2(value.data(), value.length(), _seed);
+  return static_cast<uint64_t>(hash);
+}
+
+template <typename ElementType>
+void CountingQuotientFilter<ElementType>::populate(const std::shared_ptr<const BaseSegment>& segment) {
+  resolve_segment_type<ElementType>(*segment, [&](const auto& typed_segment) {
+    auto iterable_left = create_iterable_from_segment<ElementType>(typed_segment);
+    iterable_left.for_each([&](const auto& value) {
+      if (value.is_null()) return;
+      insert(value.value());
+    });
+  });
+}
+
+template <typename ElementType>
+uint64_t CountingQuotientFilter<ElementType>::memory_consumption() const {
+  uint64_t consumption = 0;
+  boost::apply_visitor([&](auto& filter) {consumption += qf_memory_consumption(filter);}, _quotient_filter);

```

Comment:
Why `+=`? Is this called more than once?

Reply:
Fixed.